app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,49 @@
+class AlbumsController < ApplicationController
+
+  def index
+    @albums = Album.all
+  end
+
+  def new
+    @album = Album.new
+  end
+
+  def create
+    @album = Album.new(album_params)
+   if @album.save
+     redirect_to albums_path
+   else
+     render :new
+   end
+  end
+
+  def show
+    @album = Album.find(params[:id])
```

Comment:
see that you're calling `Album.find(params[:id])` multiple times in this file? I recommend something like this
```ruby
def show
  @album = get_album
end

# ...

private

def get_album
  Album.find(params[:id])
end
```